src/components/ItemsList.js

```diff
@@ -104,12 +106,28 @@ const ItemsList = () => {
                   .toLowerCase()
                   .includes(searchTerm.toLowerCase()),
               )
+              .sort((a, b) =>
+                a.daysLeftForNextPurchase > b.daysLeftForNextPurchase
```

Comment:
Love this logic! It's great and concise

Reply:
Thanks!

Reply:
I do agree it is concise. However,
- I think it would be great to transfer the callback function which you are passing to the `.sort` method  to `sortingFunctions.js` file. You can name it `sortShoppingList` and export it to `ItemsList.js` file. This makes the code less congested and more readable. Something like `.sort(sortShoppingList)` will make the code cleaner and tidy.
- This is not for you to implement but for the entire team to think about.  I think it is not efficient to filter the shopping list on every render though I could be wrong about this.  As it stands now, if a user is typing in the search bar in an attempt to filter the items, the component is re-rendered on each keystroke. This therefore means the shopping list is also being filtered on each key stroke.  I believe this could be a good use case for `useMemo` hook so that we memoize the filtered shopping list and we filter again only  when the shopping list changes in the database.
```js
const filteredShoppingList = useMemo(() => {
   if(!shoppingList) return [];
   // Do the filtering here
}, [shoppingList]);
```
Though we need to discuss and agree about this to avoid premature optimization.

Reply:
Thanks for the review and suggestion  about moving the sorting to the function file, it has now been updated. About the second point, of course! it will make sense, during the refactoring stage we could do that!